### PR TITLE
reaction_held reactions addition

### PR DESCRIPTION
**Pull request by 1sourcecontrol**
#204

### CODE REVIEW

File: Basic Rules.txt [hunk 1]
```diff
@@ -665,6 +665,18 @@ Performing a Skill Check (other than Getting Bearings/Awareness of surroundings)
 Conversing With Partners - Free Action
 Getting Your Bearings / Non Skill-Based Perception - Free Action
 
+==== Reactions & Held Actions ====
+Certain abilities or circumstances will allow you to take a special action in response to a 
+triggering event. A reaction occurs at the immediate moment that it is triggered, whether it
+is your turn or not. The most common reaction you will encounter will be from Suppression 
+(described below) or Opportunity Attacks (described above). After the reaction occurs, play
```
Comment (Turtlelord26) [inline Basic Rules.txt]: "(described above)"
I can't find that description. I see trigger circumstances but no specific definition of exactly what an opportunity attack *is*.
Do you get one attacking action? Can you use any weapon at any range? (probably not heavy weapons at melee range?) Do blindfire rules apply if you don't have your aim set on the offending character? etc.

**Review by Turtlelord26 — CHANGES_REQUESTED** [review]

File: Basic Rules.txt (hunk 1 above)
Comment (Turtlelord26) [reply Basic Rules.txt]: As a note, my requested change may not necessarily seem in-scope from the PR title but I believe that it is in-scope for the referenced issue.

File: Basic Rules.txt [hunk 2]
```diff
@@ -665,6 +665,18 @@ Performing a Skill Check (other than Getting Bearings/Awareness of surroundings)
...
+Certain abilities or circumstances will allow you to take a special action in response to a 
+triggering event. A reaction occurs at the immediate moment that it is triggered, whether it
+is your turn or not. The most common reaction you will encounter will be from Suppression 
+(described below) or Opportunity Attacks (described above). After the reaction occurs, play 
+continues as normal (i.e., if it is an opponents turn, they continue their turn). 
+	Held Actions are a type of reaction. A player may choose to forgo using up to 2 actions 
+at the end of their turn. The player must declare to their DM what those actions are to be 
+triggered by then play continues as normal. If the trigger occurs before the player's next 
+turn, the player gets to immediately use their 1 or 2 actions. If the trigger does not occur 
+before the player's turn, they lose their held action(s) and start their turn as normal.
```
Comment (bleehu) [inline Basic Rules.txt]: Let's add an example if possible.

**Review by bleehu — CHANGES_REQUESTED** [review]

**Comment by 1sourcecontrol** [comment]
Added changes. I also ended up having some scope creep and tried to more clearly define some rules leading up to Opportunity Attacks.

Please re-review @bleehu @Turtlelord26

File: Basic Rules.txt [hunk 3]
```diff
@@ -597,18 +597,27 @@ you below the minimum miss chance.
...
+	
+Opportunity Attack:
+A free attack* that is triggered when an opponent leaves melee combat or walks adjacent 
+(within melee range) to a character that is aware of their opponent. Characters 
+also incur Opportunity Attacks by performing a non-combat skill check when locked in melee 
+combat (such as trying to repair their own armor) or by trying to shoot a different opponent
+with a non-CQB weapon. 
+	*The free attack can be made with a CQB weapon or a melee weapon (depending on if the weapon
+	is already equipped in the character's hands). If the free attack is made with a CQB weapon,
+	and the weapon is not already aimed at the target, standard Hip Firing rules apply.
```
Comment (bleehu) [inline Basic Rules.txt:617]: I really like all of this. This is way more clear than it was before and I think someone who hadn't played DnD could understand how Opportunity Attacks work. Great Job!

File: Basic Rules.txt [hunk 4]
```diff
@@ -665,6 +674,28 @@ Performing a Skill Check (other than Getting Bearings/Awareness of surroundings)
...
+as normal. If the trigger occurs before the player's next turn, the player gets to immediately 
+use their 1 or 2 actions. If the trigger does not occur before the player's turn, they lose their
+held action(s) and start their turn as normal. 
+	For example, let's say my character and his/her friends are walking along a ledge. I know that
+one of the characters in the party is clumsy so I have my character move along the ledge, and I ask 
+the DM if I can save my last action to try and catch the clumsy character, should they trip. The
+DM approves my plan and play continues. On the clumsy character's turn, they start to scoot across
+the ledge and trip! As soon as they trip, my held action triggers and my character saves them from 
+falling off the ledge and their turn continues. If they had not tripped, my held action would've
+been lost at the start of my following turn.
```
Comment (bleehu) [inline Basic Rules.txt:697]: Clear, concise and creative. Fantastic example!

**Review by bleehu — APPROVED** [review]
Unfurl the sails, botswain! Shipit!

**Comment by 1sourcecontrol** [comment]
Thanks!

**Comment by Turtlelord26** [comment]
I would still like the definition of Opportunity attack to explicitly state that you get ^one Action^ of attack.

Less important, but lines 616-617 will scan better as "If the free attack is made with a CQB weapon
that is not already aimed at the target, standard Hip Firing rules apply."
Note the removal of the comma after weapon.

**Pull request merged**